include/aws/mqtt/private/fixed_header.h

```diff
@@ -59,4 +59,6 @@ AWS_MQTT_API int aws_mqtt_fixed_header_encode(struct aws_byte_buf *buf, const st
  */
 AWS_MQTT_API int aws_mqtt_fixed_header_decode(struct aws_byte_cursor *cur, struct aws_mqtt_fixed_header *header);
 
+AWS_MQTT_API int aws_mqtt311_decode_remaining_length(struct aws_byte_cursor *cur, size_t *remaining_length_out);
```

Comment:
Trivial: MQTT5's variable byte integer ( https://docs.oasis-open.org/mqtt/mqtt/v5.0/os/mqtt-v5.0-os.html#_Toc3901011 ) is encoded the same way as MQTT3's variable length encoding ( http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html#_Toc398718023 ). Just wondering if we should unify the decoding of VLIs. Not necessary but feels funny that we have two things doing the same thing differently. 

MQTT3
https://github.com/awslabs/aws-c-mqtt/blob/7aaae0ce8c644faf12fcb6544f8f105787596101/source/fixed_header.c#L32C27-L32C27

MQTT5
https://github.com/awslabs/aws-c-mqtt/blob/7aaae0ce8c644faf12fcb6544f8f105787596101/source/v5/mqtt5_decoder.c#L75

Reply:
In the future maybe, but the contracts are different and I didnt want to unify them as part of this.